salt/state.py

```diff
@@ -4056,7 +4056,8 @@ def __init__(
             mocked=False,
             loader='states',
             initial_pillar=None):
-        self.opts = opts
+        self.opts = copy.deepcopy(opts)
+        self.opts['file_client'] = 'local' if self.opts['__role'] == 'master' else self.opts['file_client']
```

Comment:
It would be good to put some inline docs here explaining why this is necessary so that we don't forget down the road.